### PR TITLE
create services and cache id from api

### PR DESCRIPTION
**Pull request by CherevatovDk**

### CODE REVIEW

File: HackerNewsWrapperApi/Controllers/StoriesController.cs [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+    {
+        _hackerHttpService = hackerHttpService;
+    }
+
     [HttpGet("best-stories")]
-    public async Task<ActionResult<IEnumerable<object>>> BestStoriesAsync(int storiesCount)
+    public async Task<ActionResult<List<StoryDto>>> BestStoriesAsync(int count)
     {
-        return Ok();
+        var s = _hackerHttpService.GetDetailsAsync();
```
Comment (dannechypor) [inline HackerNewsWrapperApi/Controllers/StoriesController.cs]: Delete this, and leave empty endpoint

File: HackerNewsWrapperApi/Program.cs [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
 builder.Services.AddControllers();
 // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
-
+builder.Services.AddMemoryCache();
+builder.Services.AddHttpClient("ApiClient", client =>
+{
+    client.BaseAddress = new Uri(builder.Configuration.GetSection("HackerApi:Url").Value!);
+
```
Comment (dannechypor) [inline HackerNewsWrapperApi/Program.cs]: Format

File: HackerNewsWrapperApi/Services/HackerHttpService.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            return value ?? new List<int>();
+        }
+
+        var response = await _httpClient.GetFromJsonAsync<List<int>>(_hackerApiSettings.GetIdsUrl());
+        _cache.Set(Constans.BestIds, response, TimeSpan.FromMinutes(5));
+        return response ?? new List<int>();
+    }
+
+    public async Task<StoryDto> GetDetailsAsync()
+    {
```
Comment (dannechypor) [inline HackerNewsWrapperApi/Services/HackerHttpService.cs]: You do not need Details in this story

File: HackerNewsWrapperApi/appsettings.json [hunk 4]
```diff
@@ -1,4 +1,8 @@
 {
+  "HackerApi": {
+    "Url": "https://hacker-news.firebaseio.com",
+    "Paths": [{"Ids":"/v0/beststories.json"},{"Item":"/v0/item/"}]
```
Comment (dannechypor) [inline HackerNewsWrapperApi/appsettings.json]: Do Paths array items like KeyValue structures

File: HackerNewsWrapperApi/Options/Helper.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
+namespace HackerNewsWrapperApi.Options;
+
+public static class Helper
+{
+    
+    public static string GetIdsUrl(this HackerApiSettings settings)
+    {
```
Comment (dannechypor) [inline HackerNewsWrapperApi/Options/Helper.cs:6]: It looks fine but refactor this to work with key value objects.

File: HackerNewsWrapperApi/Program.cs [hunk 6]
```diff
@@ -2,22 +2,16 @@
 using HackerNewsWrapperApi.Services;
 
 var builder = WebApplication.CreateBuilder(args);
-var configuration = new ConfigurationBuilder()
-    .SetBasePath(Directory.GetCurrentDirectory())
-    .AddJsonFile("appsettings.json")
-    .Build();
+new ConfigurationBuilder().SetBasePath(Directory.GetCurrentDirectory()).AddJsonFile("appsettings.json").Build();
```
Comment (Nick9-9) [inline HackerNewsWrapperApi/Program.cs]: write it in vertical structure

File: HackerNewsWrapperApi/Constans.cs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+namespace HackerNewsWrapperApi;
+
+public static class Constans
+{
```
Comment (dannechypor) [inline HackerNewsWrapperApi/Constans.cs:4]: hui hui hui

**Review by dannechypor — APPROVED** [review]

**Pull request merged**